service/src/main/scala/za/co/absa/logingw/rest/config/GitConfig.scala

```diff
@@ -18,42 +18,72 @@ package za.co.absa.logingw.rest.config
 
 import org.springframework.boot.context.properties.{ConfigurationProperties, ConstructorBinding}
 import javax.annotation.PostConstruct
-import java.io.PrintWriter
 import java.text.SimpleDateFormat
+import java.io.{PrintWriter, IOException}
+import scala.util.Try
 
 @ConstructorBinding
-@ConfigurationProperties(prefix = "logingw.rest.config")
-case class GitConfig(gitProperties: Boolean) {
+@ConfigurationProperties(prefix = "logingw.rest.config.git-info")
+case class GitConfig(generateGitProperties: Boolean, generateGitPropertiesFile: Boolean) {
 
   @PostConstruct
   def init(): Unit = {
-    if(gitProperties)
-    this.GitPropertiesGenerator.generateGitProperties()
+    if(generateGitProperties)
+      GitPropertiesGenerator.generateGitProperties(generateGitPropertiesFile)
   }
+}
+
+object GitPropertiesGenerator {
+  private var branch: String = _
+  private var commitId: String = _
+  private var commitTime: String = _
+  def generateGitProperties(writeFile: Boolean): Unit = {
+    val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+    setProperties(
+      getGitOutput("git rev-parse --abbrev-ref HEAD").getOrElse("unknown"),
+      getGitOutput("git rev-parse HEAD").getOrElse("unknown"),
+      dateFormat.format(getGitOutput("git show -s --format=%ct HEAD").map(_.toLong * 1000).getOrElse(0L))
+    )
+    if(writeFile)
+      writeGitPropertiesToFile()
+  }
+
+  def setProperties(branch: String, commitId: String, commitTime: String): Unit = {
+    this.branch = branch
+    this.commitId = commitId
+    this.commitTime = commitTime
+  }
+
+  def getBranch: String = branch
+  def getCommitId: String = commitId
+  def getCommitTime: String = commitTime
 
-  object GitPropertiesGenerator {
-    def generateGitProperties(): Unit = {
-      val gitPropertiesFile = "service\\src\\main\\resources\\git.properties"
-      val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
-      val gitBranch = sys.process.Process("git rev-parse --abbrev-ref HEAD").lineStream_!.headOption.getOrElse("unknown")
-      val gitCommitMessage = sys.process.Process("git log -1 --pretty=%B").lineStream_!.headOption.getOrElse("unknown")
-      val gitCommitId = sys.process.Process("git rev-parse HEAD").lineStream_!.headOption.getOrElse("unknown")
-      val gitCommitTime = dateFormat.format(sys.process.Process("git show -s --format=%ct HEAD").lineStream_!.headOption.map(_.toLong * 1000).getOrElse(0L))
-      val gitAuthor = sys.process.Process("git show -s --format='%ae'").lineStream_!.headOption.getOrElse("unknown")
-      val content =
-        s"""|git.branch=$gitBranch
-            |git.commit.message.full=$gitCommitMessage
-            |git.commit.id=$gitCommitId
-            |git.commit.time=$gitCommitTime
-            |git.commit.user.email=$gitAuthor
-            |""".stripMargin
```

Comment:
I see that some of the fields were omitted, but that is probably ok -- the newer version better conforms to the git info in the _simple_ form provided by the maven plugin 👍